extensions/amp-animation/0.1/web-animations.js

```diff
@@ -0,0 +1,363 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {dev, user} from '../../../src/log';
+import {getVendorJsPropertyName} from '../../../src/style';
+import {isArray, isObject} from '../../../src/types';
+import {
+  WebAnimationDef,
+  WebAnimationTimingDef,
+  WebAnimationTimingDirection,
+  WebAnimationTimingFill,
+  WebKeyframeAnimationDef,
+  WebKeyframesDef,
+  WebMultiAnimationDef,
+} from './web-animation-types';
+
+
+/**
+ * A struct for parameters for `Element.animate` call.
+ * See https://developer.mozilla.org/en-US/docs/Web/API/Element/animate
+ *
+ * @typedef {{
+ *   target: !Element,
+ *   keyframes: !WebKeyframesDef,
+ *   timing: !WebAnimationTimingDef,
+ * }}
+ */
+let InternalWebAnimationRequestDef;
+
+
+/**
+ */
+export class WebAnimationRunner {
+
+  /**
+   * @param {!Array<!InternalWebAnimationRequestDef>} requests
+   */
+  constructor(requests) {
+    /** @const @private */
+    this.requests_ = requests;
+
+    /** @private {!Array<!Animation>} */
+    this.players_ = [];
+  }
+
+  /**
+   */
+  play() {
+    this.players_ = this.requests_.map(request => {
+      return request.target.animate(request.keyframes, request.timing);
+    });
+  }
+}
+
+
+/**
+ * The scanner for the `WebAnimationDef` format. It calls the appropriate
+ * callbacks based on the discovered animation types.
+ */
+class Scanner {
+
+  /**
+   * @param {!WebAnimationDef|!Array<!WebAnimationDef>} spec
+   */
+  scan(spec) {
+    if (isArray(spec)) {
+      spec.forEach(spec => this.scan(spec));
+      return;
+    }
+
+    // WebAnimationDef: (!WebMultiAnimationDef|!WebKeyframeAnimationDef)
+    if (spec.animations) {
+      this.onMultiAnimation(/** @type {!WebMultiAnimationDef} */ (spec));
+    } else if (spec.keyframes) {
+      this.onKeyframeAnimation(/** @type {!WebKeyframeAnimationDef} */ (spec));
+    } else {
+      this.onUnknownAnimation(spec);
+    }
+  }
+
+  /** @param {!WebMultiAnimationDef} unusedSpec */
+  onMultiAnimation(unusedSpec) {
+    dev().assert(null, 'not implemented');
```

Comment:
It is. But I thought it was out pattern for `@abstract`, since `dev.assert` are removed from the binary. Is it not? Could you provide the right way?